### PR TITLE
Add Spoilers as per MSC2010

### PR DESCRIPTION
**Pull request by Sorunome**
This PR adds spoiler support as to the still WIP https://github.com/matrix-org/matrix-doc/pull/2010

Rendered example:

Signed-off-by: Sorunome <mail@sorunome.de>

### CODE REVIEW

File: src/components/views/elements/Spoiler.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    render: function() {
+        const reason = this.props.reason ? (
+            <span className="mx_EventTile_spoiler_reason">{"(" + this.props.reason + ")"}</span>
+        ) : null;
+        return (
+            <span className={"mx_EventTile_spoiler" + (this.state.visible ? " visible" : "")} onClick={this.toggleVisible.bind(this)}>
+                { reason }
+                &nbsp;
+                <span className="mx_EventTile_spoiler_content">
+                    <span dangerouslySetInnerHTML={{ __html: this.props.contentHtml }} />
```
Comment (Sorunome) [inline src/components/views/elements/Spoiler.js]: This is pretty stupid. Soru had to pass the content as HTML as, apparently, react doesn't allow attaching a DOM node as child...does anyone know a cleaner solution?

**Comment by Sorunome** [comment]
This PR is ready for review

**Comment by Sorunome** [comment]
> We'll also need signoff for the changes - depending on where you are located you might be able to get away with using a pseudonym instead of your real name, if you prefer.

Soru thought signing each commit was sufficient?

**Comment by jryans** [comment]
> Soru thought signing each commit was sufficient? As for real name, soru is not willing to use it here.

You can either sign off in commits or the PR. See the [contributing doc](https://github.com/matrix-org/matrix-js-sdk/blob/master/CONTRIBUTING.rst#sign-off) for more details. It looks like your commits have a GPG signature / verified label, but that is not the same as "sign off". Sign off means adding the text `Signed-off-by: Your Name <your@email.example.org>` to either each commit or the PR.

As for name, I believe using a name you are comfortable with identifying you is sufficient for this purpose.

Once the sign off is done, comment here and we can send this for another review round. Thanks! 😄

**Comment by Sorunome** [comment]
Okay, done, thank you!

**Comment by nadonomy** [comment]
Visually this looks fine! Are we confident in whatever blur technique we're using behaving well across all browsers?

**Comment by turt2live** [comment]
https://caniuse.com/#feat=css-filters says yes

**Comment by turt2live** [comment]
this goes to develop, so let's see how bad it can possibly be in practice :p

**Pull request merged**